src/VfacTmdb/Abstracts/Items/ItemChanges.php

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+/**
+ * This file is part of the Tmdb package.
+ *
+ * (c) Vincent Faliès <vincent@vfac.fr>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Vincent Faliès <vincent@vfac.fr>
+ * @copyright Copyright (c) 2017
+ */
+
+
+namespace VfacTmdb\Abstracts\Items;
+
+use VfacTmdb\Results;
+use VfacTmdb\Exceptions\TmdbException;
+use VfacTmdb\Interfaces\TmdbInterface;
+
+/**
+ * Abstract Item Changes class
+ * @package Tmdb
+ * @author Steve Richter <steve@nerdbra.in>
+ * @copyright Copyright (c) 2021
+ */
+abstract class ItemChanges
+{
+    /**
+     * Tmdb object
+     * @var TmdbInterface
+     */
+    protected $tmdb = null;
+    /**
+     * Logger
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger = null;
+    /**
+     * Params
+     * @var array
+     */
+    protected $params = [];
+    /**
+     * Data
+     * @var \stdClass
+     */
+    protected $data = null;
+
+    /**
+     * Constructor
+     * @param TmdbInterface $tmdb
+     * @param string $item_type (possible values: movie, person, tv, tv/season, tv/episode)
+     * @param int $item_id
+     * @param array $options
+     */
+    public function __construct(TmdbInterface $tmdb, string $item_type, int $item_id, array $options = array())
+    {
+        try {
+            $this->tmdb   = $tmdb;
+            $this->logger = $tmdb->getLogger();
+
+            $this->tmdb->checkOptionPage($options, $this->params);
+            $this->tmdb->checkOptionDateRange($options, $this->params);
+
+            $this->data = $this->tmdb->getRequest($item_type . '/' . $item_id . '/changes', $this->params);
+        } catch (TmdbException $ex) {
+            throw $ex;
+        }
+    }
+
+    /**
+     * Get Changes
+     * @return \Generator|Results\ItemChange
+     */
+    public function getChanges() : \Generator
+    {
+        if (!empty($this->data->changes)) {
+            foreach ($this->data->changes as $change) {
+                foreach ($change->items as $item) {
+                    $item->key = $change->key;
```

Comment:
The API groups changes by `key`; the part of the item which changed. Sometimes there will be multiple changes returned per key. I felt it made more sense to return each of those changes as their own ItemChange rather than an ItemChange being able to represent one or multiple changes and the user having to determine which. So we loop through each "key change" and append the key to it before creating the ItemChange.